cli/azd/pkg/tools/azcli/account.go

```diff
@@ -0,0 +1,225 @@
+package azcli
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/subscription/armsubscription"
+	"github.com/azure/azure-dev/cli/azd/pkg/convert"
+	"github.com/azure/azure-dev/cli/azd/pkg/identity"
+)
+
+type AzCliSubscriptionInfo struct {
+	Name      string `json:"name"`
+	Id        string `json:"id"`
+	TenantId  string `json:"tenantId"`
+	IsDefault bool   `json:"isDefault"`
+}
+
+type AzCliLocation struct {
+	// The human friendly name of the location (e.g. "West US 2")
+	DisplayName string `json:"displayName"`
+	// The name of the location (e.g. "westus2")
+	Name string `json:"name"`
+	// The human friendly name of the location, prefixed with a
+	// region name (e.g "(US) West US 2")
+	RegionalDisplayName string `json:"regionalDisplayName"`
+}
+
+func (cli *azCli) ListAccounts(ctx context.Context) ([]AzCliSubscriptionInfo, error) {
+	client, err := cli.createSubscriptionsClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	var rawResponse *http.Response
```

Comment:
Should we be calling `Close` on the body of the response in some cases?  I know that the caller needs to close the body of the response when using the default HTTP Client in go, so I would assume we do, but there's enough stuff going on here that I'm not sure if using `WithCaptureResponse` changes the game?

Reply:
From looking through the source the response body still appears to get closed by the standard go sdk pipeline.  We're just getting a pointer to it.

Response body is closed @ https://github.com/Azure/azure-sdk-for-go/blob/eff7206f2cc6285101103d7674f51a33d536705b/sdk/azcore/internal/exported/exported.go#L54